{"event": "config", "data": {"system": {"arch": "cuda", "sshkey": null, "nodes": [{"name": "local", "ip": "127.0.0.1", "sshport": 22, "user": "root", "main": true, "hostname": "cn-c009.server.mila.quebec", "local": true}], "self": {"name": "local", "ip": "127.0.0.1", "sshport": 22, "user": "root", "main": true, "hostname": "cn-c009.server.mila.quebec", "local": true}}, "dirs": {"base": "/network/scratch/o/ortizgas/data/milabench", "venv": "/network/scratch/o/ortizgas/data/milabench/venv/torch", "data": "/network/scratch/o/ortizgas/data/milabench/data", "runs": "/network/scratch/o/ortizgas/data/milabench/runs", "extra": "/network/scratch/o/ortizgas/data/milabench/extra/rlhf-gpus", "cache": "/network/scratch/o/ortizgas/data/milabench/cache"}, "group": "rlhf-gpus", "install_group": "torch", "install_variant": "cuda", "run_name": "Quadro-RTX-8000_rlhf-gpus_192.staging", "enabled": true, "capabilities": {"nodes": 1}, "max_duration": 600, "voir": {"options": {"stop": 60, "interval": "1s"}}, "validation": {"usage": {"gpu_load_threshold": 0.5, "gpu_mem_threshold": 0.5}}, "config_base": "/home/mila/o/ortizgas/CODE/milabench/config", "config_file": "/home/mila/o/ortizgas/CODE/milabench/config/standard.yaml", "definition": "/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf", "plan": {"method": "njobs", "n": 1}, "tags": ["llm", "multigpu", "rl", "rlhf"], "argv": {"--output_dir": "{milabench_extra}/output", "--model_name_or_path": "EleutherAI/pythia-1b-deduped", "--per_device_train_batch_size": 64, "--logging_strategy": "no", "--log_level": "critical", "--bf16": true}, "weight": 1.0, "name": "rlhf-gpus", "tag": ["rlhf-gpus", "0"], "job-number": 0, "devices": [0, 1, 2, 3]}, "pipe": null}
{"event": "meta", "data": {"cpu": {"count": 64, "brand": "AMD EPYC 7502 32-Core Processor"}, "os": {"sysname": "Linux", "nodename": "cn-c009.server.mila.quebec", "release": "5.15.0-101-generic", "version": "#111-Ubuntu SMP Tue Mar 5 20:16:58 UTC 2024", "machine": "x86_64"}, "accelerators": {"arch": "cuda", "gpus": {"0": {"minor_number": "4", "device": 0, "product": "Quadro RTX 8000", "memory": {"used": 541.9375, "total": 46080.0}, "utilization": {"compute": 0.0, "memory": 0}, "temperature": 42.0, "power": 26.908, "selection_variable": "CUDA_VISIBLE_DEVICES"}, "1": {"minor_number": "5", "device": 1, "product": "Quadro RTX 8000", "memory": {"used": 541.9375, "total": 46080.0}, "utilization": {"compute": 0.0, "memory": 0}, "temperature": 39.0, "power": 28.269, "selection_variable": "CUDA_VISIBLE_DEVICES"}, "2": {"minor_number": "6", "device": 2, "product": "Quadro RTX 8000", "memory": {"used": 541.9375, "total": 46080.0}, "utilization": {"compute": 0.0, "memory": 0}, "temperature": 45.0, "power": 25.817, "selection_variable": "CUDA_VISIBLE_DEVICES"}, "3": {"minor_number": "7", "device": 3, "product": "Quadro RTX 8000", "memory": {"used": 541.9375, "total": 46080.0}, "utilization": {"compute": 0.0, "memory": 0}, "temperature": 38.0, "power": 26.708, "selection_variable": "CUDA_VISIBLE_DEVICES"}}}, "date": 1733966274.936317, "milabench": {"tag": "v1.0.0_RC1-13-g7a90b16", "commit": "7a90b1691650232ecd63abded0b7be84bb294c05", "date": "2024-11-08 11:28:56 -0500"}, "pytorch": {"torch": "2.4.0+cu121", "compiler": "GCC 9.3", "cpp": "C++ Version: 201703", "intel": "Intel(R) oneAPI Math Kernel Library Version 2022.2-Product Build 20220804 for Intel(R) 64 architecture applications", "mkl": "OpenMP 201511 (a.k.a. OpenMP 4.5)", "openmp": "OpenMP 201511 (a.k.a. OpenMP 4.5)", "lapack": "LAPACK is enabled (usually provided by MKL)", "nnpack": "NNPACK is enabled", "cpu": "CPU capability usage: AVX2", "build_settings": {"BLAS_INFO": "mkl", "BUILD_TYPE": "Release", "CUDA_VERSION": "12.1", "CUDNN_VERSION": "9.1.0", "CXX_COMPILER": "/opt/rh/devtoolset-9/root/usr/bin/c++", "CXX_FLAGS": "-D_GLIBCXX_USE_CXX11_ABI=0 -fabi-version=11 -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOROCTRACER -DUSE_FBGEMM -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Werror=bool-operation -Wnarrowing -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-stringop-overflow -Wsuggest-override -Wno-psabi -Wno-error=pedantic -Wno-error=old-style-cast -Wno-missing-braces -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow", "LAPACK_INFO": "mkl", "PERF_WITH_AVX": "1", "PERF_WITH_AVX2": "1", "PERF_WITH_AVX512": "1", "TORCH_VERSION": "2.4.0", "USE_CUDA": "ON", "USE_CUDNN": "ON", "USE_CUSPARSELT": "1", "USE_EXCEPTION_PTR": "1", "USE_GFLAGS": "OFF", "USE_GLOG": "OFF", "USE_GLOO": "ON", "USE_MKL": "ON", "USE_MKLDNN": "ON", "USE_MPI": "OFF", "USE_NCCL": "1", "USE_NNPACK": "ON", "USE_OPENMP": "ON", "USE_ROCM": "OFF", "USE_ROCM_KERNEL_ASSERT": "OFF"}}}, "pipe": null}
{"event": "start", "data": {"command": ["/home/mila/o/ortizgas/CODE/milabench/milabench/scripts/activator", "/network/scratch/o/ortizgas/data/milabench/venv/torch", "/network/scratch/o/ortizgas/data/milabench/cache", "accelerate", "launch", "--mixed_precision=bf16", "--dynamo_backend=no", "--machine_rank=0", "--num_machines=1", "--multi_gpu", "--gradient_accumulation_steps=1", "--num_cpu_threads_per_process=4", "--main_process_ip=127.0.0.1", "--main_process_port=29400", "--num_processes=4", "/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py", "--output_dir", "/network/scratch/o/ortizgas/data/milabench/extra/rlhf-gpus/output", "--model_name_or_path", "EleutherAI/pythia-1b-deduped", "--per_device_train_batch_size", "192", "--logging_strategy", "no", "--log_level", "critical", "--bf16"], "time": 1733948275.0283494}, "pipe": null}
{"event": "line", "data": "/network/scratch/o/ortizgas/data/milabench/cache\n", "pipe": "stdout"}
{"event": "line", "data": "/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884\n", "pipe": "stderr"}
{"event": "line", "data": "  warnings.warn(\n", "pipe": "stderr"}
{"event": "line", "data": "/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884\n", "pipe": "stderr"}
{"event": "line", "data": "  warnings.warn(\n", "pipe": "stderr"}
{"event": "line", "data": "/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884\n", "pipe": "stderr"}
{"event": "line", "data": "  warnings.warn(\n", "pipe": "stderr"}
{"event": "line", "data": "/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884\n", "pipe": "stderr"}
{"event": "line", "data": "  warnings.warn(\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "No overseer found\n", "pipe": "stdout"}
{"event": "line", "data": "No overseer found\n", "pipe": "stdout"}
{"event": "line", "data": "===training policy===\n", "pipe": "stdout"}
{"event": "line", "data": "\r  0%|          | 0/22 [00:00<?, ?it/s]", "pipe": "stderr"}
{"event": "data", "data": {"progress": [0, 70], "task": "early_stop"}, "pipe": "data"}
{"event": "line", "data": "No overseer found\n", "pipe": "stdout"}
{"event": "line", "data": "No overseer found\n", "pipe": "stdout"}
{"event": "data", "data": {"task": "main", "time": 1733948305.7299078, "gpudata": {"0": {"memory": [5050.8125, 46080.0], "load": 0.0, "temperature": 42.0, "power": 70.393}, "1": {"memory": [4562.8125, 46080.0], "load": 0.0, "temperature": 39.0, "power": 72.329}, "2": {"memory": [4562.8125, 46080.0], "load": 0.0, "temperature": 45.0, "power": 67.999}, "3": {"memory": [4542.8125, 46080.0], "load": 0.0, "temperature": 38.0, "power": 65.652}}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733948305.7299078, "iodata": {"read_count": 0, "write_count": 70, "read_bytes": 0, "read_time": 0, "write_time": 9, "busy_time": 84}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733948305.7299078, "netdata": {"bytes_sent": 816920862178, "bytes_recv": 797771638287, "packets_sent": 1396948140, "packets_recv": 1407918694, "errin": 0, "errout": 0, "dropin": 3317733, "dropout": 0}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733948305.7299078, "cpudata": {"memory": [14138703872, 405580578816], "load": 0.0}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733948315.7503598, "gpudata": {"0": {"memory": [43668.8125, 46080.0], "load": 1.0, "temperature": 50.0, "power": 229.212}, "1": {"memory": [45512.8125, 46080.0], "load": 1.0, "temperature": 46.0, "power": 233.356}, "2": {"memory": [43904.8125, 46080.0], "load": 1.0, "temperature": 55.0, "power": 230.674}, "3": {"memory": [45492.8125, 46080.0], "load": 1.0, "temperature": 47.0, "power": 224.047}}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733948315.7503598, "iodata": {"read_count": 2, "write_count": 248, "read_bytes": 8192, "read_time": 0, "write_time": 31, "busy_time": 224}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733948315.7503598, "netdata": {"bytes_sent": 816920937046, "bytes_recv": 797771653055, "packets_sent": 1396948245, "packets_recv": 1407918764, "errin": 0, "errout": 0, "dropin": 3317733, "dropout": 0}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733948315.7503598, "cpudata": {"memory": [15080280064, 405580578816], "load": 7.2}}, "pipe": "data"}
{"event": "line", "data": "[rank1]: Traceback (most recent call last):\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 133, in <module>\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     main()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 124, in main\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     trainer.train()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/trl/trainer/ppov2_trainer.py\", line 465, in train\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     accelerator.backward(loss)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/accelerator.py\", line 2196, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     loss.backward(**kwargs)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/_tensor.py\", line 521, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     torch.autograd.backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/__init__.py\", line 289, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     _engine_run_backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/graph.py\", line 768, in _engine_run_backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]: torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 1.91 GiB. GPU 1 has a total capacity of 44.47 GiB of which 1.88 GiB is free. Including non-PyTorch memory, this process has 42.59 GiB memory in use. Of the allocated memory 38.60 GiB is allocated by PyTorch, and 3.66 GiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]: Traceback (most recent call last):\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 133, in <module>\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     main()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 124, in main\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     trainer.train()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/trl/trainer/ppov2_trainer.py\", line 465, in train\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     accelerator.backward(loss)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/accelerator.py\", line 2196, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     loss.backward(**kwargs)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/_tensor.py\", line 521, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     torch.autograd.backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/__init__.py\", line 289, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     _engine_run_backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/graph.py\", line 768, in _engine_run_backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]: torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 1.91 GiB. GPU 3 has a total capacity of 44.47 GiB of which 1.90 GiB is free. Including non-PyTorch memory, this process has 42.57 GiB memory in use. Of the allocated memory 38.60 GiB is allocated by PyTorch, and 3.66 GiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]: Traceback (most recent call last):\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 133, in <module>\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     main()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 124, in main\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     trainer.train()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/trl/trainer/ppov2_trainer.py\", line 465, in train\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     accelerator.backward(loss)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/accelerator.py\", line 2196, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     loss.backward(**kwargs)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/_tensor.py\", line 521, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     torch.autograd.backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/__init__.py\", line 289, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     _engine_run_backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/graph.py\", line 768, in _engine_run_backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]: torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 1.91 GiB. GPU 0 has a total capacity of 44.47 GiB of which 1.50 GiB is free. Process 1132243 has 162.00 MiB memory in use. Process 1132242 has 162.00 MiB memory in use. Process 1132241 has 162.00 MiB memory in use. Including non-PyTorch memory, this process has 42.49 GiB memory in use. Of the allocated memory 40.83 GiB is allocated by PyTorch, and 1.33 GiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]: Traceback (most recent call last):\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 133, in <module>\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     main()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 124, in main\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     trainer.train()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/trl/trainer/ppov2_trainer.py\", line 465, in train\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     accelerator.backward(loss)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/accelerator.py\", line 2196, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     loss.backward(**kwargs)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/_tensor.py\", line 521, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     torch.autograd.backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/__init__.py\", line 289, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     _engine_run_backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/graph.py\", line 768, in _engine_run_backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]: torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 1.91 GiB. GPU 2 has a total capacity of 44.47 GiB of which 1.76 GiB is free. Including non-PyTorch memory, this process has 42.71 GiB memory in use. Of the allocated memory 40.98 GiB is allocated by PyTorch, and 1.39 GiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)\n", "pipe": "stderr"}
{"event": "line", "data": "\r  0%|          | 0/22 [00:17<?, ?it/s]", "pipe": "stderr"}
{"event": "line", "data": "\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:[W1211 15:18:41.488468384 ProcessGroupNCCL.cpp:1168] Warning: WARNING: process group has NOT been destroyed before we destruct ProcessGroupNCCL. On normal program exit, the application should call destroy_process_group to ensure that any pending NCCL operations have finished in this process. In rare cases this process can exit before this point and block the progress of another member of the process group. This constraint has always been present,  but this warning has only been added since PyTorch 2.4 (function operator())\n", "pipe": "stderr"}
{"event": "line", "data": "W1211 15:18:42.074319 140074219791488 torch/distributed/elastic/multiprocessing/api.py:858] Sending process 1132240 closing signal SIGTERM\n", "pipe": "stderr"}
{"event": "line", "data": "W1211 15:18:42.074621 140074219791488 torch/distributed/elastic/multiprocessing/api.py:858] Sending process 1132242 closing signal SIGTERM\n", "pipe": "stderr"}
{"event": "line", "data": "W1211 15:18:42.074717 140074219791488 torch/distributed/elastic/multiprocessing/api.py:858] Sending process 1132243 closing signal SIGTERM\n", "pipe": "stderr"}
{"event": "line", "data": "E1211 15:18:42.904245 140074219791488 torch/distributed/elastic/multiprocessing/api.py:833] failed (exitcode: 1) local_rank: 1 (pid: 1132241) of binary: /network/scratch/o/ortizgas/data/milabench/venv/torch/bin/python\n", "pipe": "stderr"}
{"event": "line", "data": "Traceback (most recent call last):\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/bin/accelerate\", line 8, in <module>\n", "pipe": "stderr"}
{"event": "line", "data": "    sys.exit(main())\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/commands/accelerate_cli.py\", line 48, in main\n", "pipe": "stderr"}
{"event": "line", "data": "    args.func(args)\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/commands/launch.py\", line 1165, in launch_command\n", "pipe": "stderr"}
{"event": "line", "data": "    multi_gpu_launcher(args)\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/commands/launch.py\", line 799, in multi_gpu_launcher\n", "pipe": "stderr"}
{"event": "line", "data": "    distrib_run.run(args)\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/distributed/run.py\", line 892, in run\n", "pipe": "stderr"}
{"event": "line", "data": "    elastic_launch(\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/distributed/launcher/api.py\", line 133, in __call__\n", "pipe": "stderr"}
{"event": "line", "data": "    return launch_agent(self._config, self._entrypoint, list(args))\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/distributed/launcher/api.py\", line 264, in launch_agent\n", "pipe": "stderr"}
{"event": "line", "data": "    raise ChildFailedError(\n", "pipe": "stderr"}
{"event": "line", "data": "torch.distributed.elastic.multiprocessing.errors.ChildFailedError: ", "pipe": "stderr"}
{"event": "line", "data": "\n", "pipe": "stderr"}
{"event": "line", "data": "============================================================\n", "pipe": "stderr"}
{"event": "line", "data": "/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py FAILED\n", "pipe": "stderr"}
{"event": "line", "data": "------------------------------------------------------------\n", "pipe": "stderr"}
{"event": "line", "data": "Failures:\n", "pipe": "stderr"}
{"event": "line", "data": "  <NO_OTHER_FAILURES>\n", "pipe": "stderr"}
{"event": "line", "data": "------------------------------------------------------------\n", "pipe": "stderr"}
{"event": "line", "data": "Root Cause (first observed failure):\n", "pipe": "stderr"}
{"event": "line", "data": "[0]:\n", "pipe": "stderr"}
{"event": "line", "data": "  time      : 2024-12-11_15:18:42\n", "pipe": "stderr"}
{"event": "line", "data": "  host      : cn-c009.server.mila.quebec\n", "pipe": "stderr"}
{"event": "line", "data": "  rank      : 1 (local_rank: 1)\n", "pipe": "stderr"}
{"event": "line", "data": "  exitcode  : 1 (pid: 1132241)\n", "pipe": "stderr"}
{"event": "line", "data": "  error_file: <N/A>\n", "pipe": "stderr"}
{"event": "line", "data": "  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html\n", "pipe": "stderr"}
{"event": "line", "data": "============================================================\n", "pipe": "stderr"}
{"event": "end", "data": {"command": ["/home/mila/o/ortizgas/CODE/milabench/milabench/scripts/activator", "/network/scratch/o/ortizgas/data/milabench/venv/torch", "/network/scratch/o/ortizgas/data/milabench/cache", "accelerate", "launch", "--mixed_precision=bf16", "--dynamo_backend=no", "--machine_rank=0", "--num_machines=1", "--multi_gpu", "--gradient_accumulation_steps=1", "--num_cpu_threads_per_process=4", "--main_process_ip=127.0.0.1", "--main_process_port=29400", "--num_processes=4", "/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py", "--output_dir", "/network/scratch/o/ortizgas/data/milabench/extra/rlhf-gpus/output", "--model_name_or_path", "EleutherAI/pythia-1b-deduped", "--per_device_train_batch_size", "192", "--logging_strategy", "no", "--log_level", "critical", "--bf16"], "time": 1733948323.602925, "return_code": 1}, "pipe": null}
